packages/web3-eth-accounts/src/common/utils.ts

```diff
@@ -0,0 +1,619 @@
+/**
+ * Parses a genesis.json exported from Geth into parameters for Common instance
+ * @param json representing the Geth genesis file
+ * @param name optional chain name
+ * @returns parsed params
+ */
+export function parseGethGenesis(json: any, name?: string, mergeForkIdPostMerge?: boolean) {
+	try {
+		if (['config', 'difficulty', 'gasLimit', 'alloc'].some(field => !(field in json))) {
+			throw new Error('Invalid format, expected geth genesis fields missing');
+		}
+		if (name !== undefined) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, no-param-reassign
+			json.name = name;
+		}
+		return parseGethParams(json, mergeForkIdPostMerge);
+	} catch (e: any) {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/restrict-template-expressions
+		throw new Error(`Error parsing parameters file: ${e.message}`);
+	}
+}
+
+/**
+ * Pads a `String` to have an even length
+ * @param value
+ * @return output
+ */
+export function padToEven(value: string): string {
+	let a = value;
+
+	if (typeof a !== 'string') {
+		throw new Error(`[padToEven] value must be type 'string', received ${typeof a}`);
+	}
+
+	if (a.length % 2) a = `0${a}`;
+
+	return a;
+}
+
+/**
+ * Converts an `Number` to a `Buffer`
+ * @param {Number} i
+ * @return {Buffer}
+ */
+export const intToBuffer = function (i: number) {
+	const hex = intToHex(i);
+	return Buffer.from(padToEven(hex.slice(2)), 'hex');
+};
+
+/**
+ * Attempts to turn a value into a `Buffer`.
+ * Inputs supported: `Buffer`, `String` (hex-prefixed), `Number`, null/undefined, `BigInt` and other objects
+ * with a `toArray()` or `toBuffer()` method.
+ * @param v the value
+ */
+export const toBuffer = function (v: ToBufferInputTypes): Buffer {
+	// eslint-disable-next-line no-null/no-null
+	if (v === null || v === undefined) {
+		return Buffer.allocUnsafe(0);
+	}
+
+	if (Buffer.isBuffer(v)) {
+		return Buffer.from(v);
+	}
+
+	if (Array.isArray(v) || v instanceof Uint8Array) {
+		return Buffer.from(v as Uint8Array);
+	}
+
+	if (typeof v === 'string') {
+		if (!isHexString(v)) {
+			throw new Error(
+				`Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: ${v}`,
+			);
+		}
+		return Buffer.from(padToEven(stripHexPrefix(v)), 'hex');
+	}
+
+	if (typeof v === 'number') {
+		return intToBuffer(v);
+	}
+
+	if (typeof v === 'bigint') {
+		if (v < BigInt(0)) {
+			throw new Error(`Cannot convert negative bigint to buffer. Given: ${v}`);
+		}
+		let n = v.toString(16);
+		if (n.length % 2) n = `0${n}`;
+		return Buffer.from(n, 'hex');
+	}
+
+	if (v.toArray) {
+		// converts a BN to a Buffer
+		return Buffer.from(v.toArray());
+	}
+
+	if (v.toBuffer) {
+		return Buffer.from(v.toBuffer());
+	}
+
+	throw new Error('invalid type');
+};
+
+/**
+ * Converts a `Buffer` into a `0x`-prefixed hex `String`.
+ * @param buf `Buffer` object to convert
+ */
+export const bufferToHex = function (_buf: Buffer): string {
```

Comment:
We have [bytesToHex](https://github.com/web3/web3.js/blob/de5617ce4f0f50785e075c7638deeac1d6ed615d/packages/web3-utils/src/converters.ts#L127) method

Reply:
here is different input param type `byte` and `Buffer`